test/sets.jl

```diff
@@ -492,65 +492,61 @@ end
-        for count = (1, 0x1, big(1))
-            @test replace(x->x.second == 2, d, 0=>0, count=count) in [Dict(1=>2, 0=>0),
-                                                                      Dict(2=>2, 0=>0)]
-        end
-        s = Set([1, 2, 3])
-        @test replace(x->maybe(2x, x>1), s) == Set([1, 4, 6])
-        for count = (1, 0x1, big(1))
-            @test replace(x->maybe(2x, x>1), s, count=count) in [Set([1, 4, 3]), Set([1, 2, 6])]
-        end
-        @test replace(s, 1=>4) == Set([2, 3, 4])
-        @test replace!(s, 1=>2) === s
-        @test s == Set([2, 3])
-        @test replace!(x->2x, s, count=0x1) in [Set([4, 3]), Set([2, 6])]
+    a = [1, 2, 3, 1]
+    @test replace(x -> iseven(x) ? 2x : x, a) == [1, 4, 3, 1]
+    @test replace!(x -> iseven(x) ? 2x : x, a) === a
+    @test a == [1, 4, 3, 1]
+    @test replace(a, 1=>0) == [0, 4, 3, 0]
+    for count = (1, 0x1, big(1))
+        @test replace(a, 1=>0, count=count) == [0, 4, 3, 1]
+    end
+    @test replace!(a, 1=>2) === a
+    @test a == [2, 4, 3, 2]
+    @test replace!(x->2x, a, count=0x2) == [4, 8, 3, 2]
+
+    d = Dict(1=>2, 3=>4)
+    @test replace(x->x.first > 2, d, 0=>0) == Dict(1=>2, 0=>0)
+    @test replace!(x -> x.first > 2 ? x.first=>2*x.second : x, d) === d
+    @test d == Dict(1=>2, 3=>8)
+    @test replace(d, (3=>8)=>(0=>0)) == Dict(1=>2, 0=>0)
+    @test replace!(d, (3=>8)=>(2=>2)) === d
+    @test d == Dict(1=>2, 2=>2)
+    for count = (1, 0x1, big(1))
+        @test replace(x->x.second == 2, d, 0=>0, count=count) in [Dict(1=>2, 0=>0),
+                                                                  Dict(2=>2, 0=>0)]
+    end
+    s = Set([1, 2, 3])
+    @test replace(x -> x > 1 ? 2x : x, s) == Set([1, 4, 6])
+    for count = (1, 0x1, big(1))
+        @test replace(x -> x > 1 ? 2x : x, s, count=count) in [Set([1, 4, 3]), Set([1, 2, 6])]
+    end
+    @test replace(s, 1=>4) == Set([2, 3, 4])
+    @test replace!(s, 1=>2) === s
+    @test s == Set([2, 3])
+    @test replace!(x->2x, s, count=0x1) in [Set([4, 3]), Set([2, 6])]
 
-        for count = (0, 0x0, big(0))
-            @test replace([1, 2], 1=>0, 2=>0, count=count) == [1, 2] # count=0 --> no replacements
-        end
+    for count = (0, 0x0, big(0))
+        @test replace([1, 2], 1=>0, 2=>0, count=count) == [1, 2] # count=0 --> no replacements
     end
+
     # test collisions with AbstractSet/AbstractDict
     @test replace!(x->2x, Set([3, 6])) == Set([6, 12])
     @test replace!(x->2x, Set([1:20;])) == Set([2:2:40;])
     @test replace!(kv -> (2kv[1] => kv[2]), Dict(1=>2, 2=>4, 4=>8, 8=>16)) == Dict(2=>2, 4=>4, 8=>8, 16=>16)
 
-    # test Some(nothing)
-    a = [1, 2, nothing, 4]
-    @test replace(x -> x === nothing ? 0 : Some(nothing), a) == [nothing, nothing, 0, nothing]
-    @test replace(x -> x === nothing ? 0 : nothing, a) == [1, 2, 0, 4]
-    @test replace!(x -> x !== nothing ? Some(nothing) : nothing, a) == [nothing, nothing, nothing, nothing]
+    # test nothing
```

Comment:
Is it useful to keep these tests now that `nothing` is no longer special? Otherwise, looks good!

Reply:
I hesitated and thought it doesn't hurt to keep those, but I now think it's better to shrink these (maybe keep a couple such tests to be sure `nothing` is not treated specially).